### ISSUE TITLE
ERROR: Invalid requirement: 'torch==1.9.0+cu111

### ISSUE THREAD

**Issue by Rotoslider** [SKrisanski/FSCT#15]
(FSCT) C:\Users\me\Documents\FSCT>pip install -r requirements.txt
ERROR: Invalid requirement: 'torch==1.9.0+cu111--find-links https://download.pytorch.org/whl/torch_stable.html' (from line 1 of requirements.txt)

latest version is:
conda install pytorch torchvision torchaudio cudatoolkit=11.3 -c pytorch

could you update the requirements.txt so this will build. I installed the latest version above but now get a build torch-sparse error

**Comment by Rotoslider** [comment]
This is what I changed the requirements.txt file to allow it to build on a windows 10 x64 machine 
`--find-links https://download.pytorch.org/whl/torch_stable.html
--find-links https://pytorch-geometric.com/whl/torch-1.11.0%2Bcu113.html

certifi==2021.10.8
charset-normalizer==2.0.12
colorama==0.4.4
cycler==0.11.0
Cython==0.29.28
fonttools==4.33.3
googledrivedownloader==0.4
hdbscan==0.8.28
idna==3.3
imageio==2.19.0
importlib-metadata==4.11.3
isodate==0.6.1
Jinja2==3.1.2
joblib==1.1.0
kiwisolver==1.4.2
laspy==2.1.2
Markdown==3.3.7
MarkupSafe==2.1.1
matplotlib==3.5.2
mdutils==1.3.1
networkx==2.8
numpy==1.22.3
packaging==21.3
pandas==1.4.2
Pillow==9.1.0
pyparsing==3.0.8
python-dateutil==2.8.2
python-louvain==0.16
pytz==2022.1
PyWavelets==1.3.0
rdflib==6.1.1
requests==2.27.1
scikit-image==0.19.2
scikit-learn==1.0.2
scikit-spatial==6.4.0
scipy==1.8.0
six==1.16.0
sklearn==0.0
threadpoolctl==3.1.0
tifffile==2022.5.4
torch==1.11+cu113
torch-cluster==1.6.0
torch-geometric==2.0.4
torch-scatter==2.0.9
torch-sparse==0.6.13
torch-spline-conv==1.2.1
tqdm==4.64.0
typing_extensions==4.2.0
urllib3==1.26.9
zipp==3.8.0
`

**Comment by SKrisanski** [comment]
Thanks for letting me know! I'll add a fix in the near future.

**Comment by Rotoslider** [comment]
Here is a list of the current environment.  There may be some in there that are newer than the txt file I forgot to add.
And thanks for making FSCT available to the public. So far working good. Best segmentation I have found so far. 
# $ conda create --name <env> --file <this file>
# platform: win-64
@EXPLICIT
https://repo.anaconda.com/pkgs/main/win-64/blas-1.0-mkl.conda
https://repo.anaconda.com/pkgs/main/win-64/ca-certificates-2022.4.26-haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/intel-openmp-2021.4.0-haa95532_3556.conda
https://conda.anaconda.org/pytorch/noarch/pytorch-mutex-1.0-cuda.tar.bz2
https://repo.anaconda.com/pkgs/main/noarch/tzdata-2022a-hda174b7_0.conda
https://repo.anaconda.com/pkgs/main/win-64/vs2015_runtime-14.27.29016-h5e58377_2.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl-2021.4.0-haa95532_640.conda
https://repo.anaconda.com/pkgs/main/win-64/vc-14.2-h21ff451_1.conda
https://repo.anaconda.com/pkgs/main/win-64/bzip2-1.0.8-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/cudatoolkit-11.3.1-h59b6b97_2.conda
https://repo.anaconda.com/pkgs/main/win-64/jpeg-9e-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libffi-3.4.2-h604cdb4_1.conda
https://repo.anaconda.com/pkgs/main/win-64/libuv-1.40.0-he774522_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libwebp-1.2.2-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/lz4-c-1.9.3-h2bbff1b_1.conda
https://repo.anaconda.com/pkgs/main/win-64/openssl-1.1.1o-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/sqlite-3.38.3-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/tk-8.6.12-h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/xz-5.2.5-h8cc25b3_1.conda
https://repo.anaconda.com/pkgs/main/win-64/zlib-1.2.12-h8cc25b3_2.conda
https://repo.anaconda.com/pkgs/main/win-64/libpng-1.6.37-h2a8f88b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/python-3.10.4-hbb2ffb3_0.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/zstd-1.5.2-h19a0ad4_0.conda
https://repo.anaconda.com/pkgs/main/win-64/certifi-2022.5.18.1-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/noarch/charset-normalizer-2.0.4-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/win-64/freetype-2.10.4-hd328e21_0.conda
https://repo.anaconda.com/pkgs/main/noarch/idna-3.3-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/win-64/libtiff-4.2.0-he0120a3_1.conda
https://repo.anaconda.com/pkgs/main/noarch/pycparser-2.21-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/noarch/six-1.16.0-pyhd3eb1b0_1.conda
https://repo.anaconda.com/pkgs/main/noarch/typing_extensions-4.1.1-pyh06a4308_0.conda
https://repo.anaconda.com/pkgs/main/noarch/wheel-0.37.1-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/win-64/win_inet_pton-1.1.0-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/wincertstore-0.2-py310haa95532_2.conda
https://repo.anaconda.com/pkgs/main/win-64/cffi-1.15.0-py310h2bbff1b_1.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl-service-2.4.0-py310h2bbff1b_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pillow-9.0.1-py310hdc2b20a_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pysocks-1.7.1-py310haa95532_0.conda
https://conda.anaconda.org/pytorch/win-64/pytorch-1.11.0-py3.10_cuda11.3_cudnn8_0.tar.bz2
https://repo.anaconda.com/pkgs/main/win-64/setuptools-61.2.0-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/win-64/brotlipy-0.7.0-py310h2bbff1b_1002.conda
https://repo.anaconda.com/pkgs/main/win-64/cryptography-37.0.1-py310h21b164f_0.conda
https://repo.anaconda.com/pkgs/main/win-64/numpy-base-1.22.3-py310h206c741_0.conda
https://repo.anaconda.com/pkgs/main/win-64/pip-21.2.4-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/noarch/pyopenssl-22.0.0-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/win-64/urllib3-1.26.9-py310haa95532_0.conda
https://repo.anaconda.com/pkgs/main/noarch/requests-2.27.1-pyhd3eb1b0_0.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl_fft-1.3.1-py310ha0764ea_0.conda
https://repo.anaconda.com/pkgs/main/win-64/mkl_random-1.2.2-py310h4ed8f06_0.conda
https://repo.anaconda.com/pkgs/main/win-64/numpy-1.22.3-py310h6d2d95c_0.conda
https://conda.anaconda.org/pytorch/win-64/torchaudio-0.11.0-py310_cu113.tar.bz2
https://conda.anaconda.org/pytorch/win-64/torchvision-0.12.0-py310_cu113.tar.bz2

**Comment by Rotoslider** [comment]
This list also works well with Ubuntu 20.04.
